rtos/Thread.cpp

```diff
@@ -264,7 +264,7 @@ uint32_t Thread::free_stack() {
 
 #if defined(MBED_OS_BACKEND_RTX5)
     if (_tid != NULL) {
-        os_thread_t *thread = (os_thread_t *)_tid;
+        mbed_rtos_storage_thread_t *thread = (mbed_rtos_storage_thread_t *)_tid;
```

Comment:
isn't os_thread_t a public type? The storage object should be only used for storage backing.

Reply:
It's defined in `rtx_lib.h` that is not public. I'll review this locally with @JonatanAntoni once he's back

Reply:
Use defines from `rtx_os.h` instead, i.e. `osRtx*_t`.